components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -311,11 +342,14 @@ export default class SidebarCategory extends React.PureComponent<Props, State> {
                                                 draggingOverDMCategory: droppableSnapshot.isDraggingOver && category.type === CategoryTypes.DIRECT_MESSAGES,
                                             })}
                                         >
-                                            <div className='SidebarChannelGroupHeader'>
+                                            <div
+                                                className={classNames('SidebarChannelGroupHeader', {
+                                                    draggingOver: droppableSnapshot.isDraggingOver && category.type !== CategoryTypes.DIRECT_MESSAGES && !(isNewCategory && (!channels || !channels.length)),
```

Comment:
I'm not sure I follow, what does this solve? This class is applied so that the header shows a box shadow when the header itself is being dragged over, and we need it turned off for direct messages. 

It also doesn't do this for `isNewCategory` since we didn't want the box shadow to show there so that the user would be more focused towards the actual drop zone rather than the header.

I can add a comment to clarify this, but I'm not sure how splitting up classes here changes.

Reply:
I was thinking it would make it clearer what this code is doing like naming parts of an if statement.
```
droppableSnapshot.isDraggingOver && category.type !== CategoryTypes.DIRECT_MESSAGES && !(isNewCategory && (!channels || !channels.length))
```
is incredibly confusing, especially with all the nested not statements, but if we were to break it up a bit and maybe name some parts of it, it would be much easier to follow.

I'm trying to come up with an example, but I'm honestly having trouble explaining when exactly that class gets applied.

Reply:
Spoke with Harrison, I've simplified the condition for easier readability.